src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -42,21 +42,25 @@ namespace ErrorCodes
     extern const int ARGUMENT_OUT_OF_BOUND;
 }
 
+static size_t chooseBufferSize(const ReadSettings & settings, size_t file_size)
+{
+    /// Buffers used for prefetch or pre-download better to have enough size, but not bigger than the whole file.
+    return std::min<size_t>(std::max<size_t>(settings.prefetch_buffer_size, DBMS_DEFAULT_BUFFER_SIZE), file_size);
```

Comment:
is max with DBMS_DEFAULT_BUFFER_SIZE necessary? seems redundant, if user makes the buffer too small it's his fault? as by default prefetch_buffer_size is equal to DBMS_DEFAULT_BUFFER_SIZE then it should be ok

Reply:
https://github.com/ClickHouse/ClickHouse/blob/3dfc0bd26597ca8c9e6c30343470de15553f3ac3/src/Storages/MergeTree/MergeTreeReaderStream.cpp#L87
we adjust buffer size (including prefetch buffer) based on mark range sizes. they could be fairly small (few KBs).
since we use them to download segments that (almost) cannot be smaller than 1MB I think it makes sense to enforce this lower bound.

Reply:
yes, good point